internal/codegen/databasecodegen/readtransaction.go

```diff
@@ -400,7 +402,7 @@ func (g ReadTransactionCodeGenerator) generateReadInterleavedRowsMethod(f *codeg
 		parentName := strcase.UpperCamelCase(string(parent.Name))
 		row := RowCodeGenerator{Table: child}
 		childName := strcase.UpperCamelCase(string(child.Name))
-		f.P("if query.", childName, " {")
+		f.P("if query.", childName, " && !", reflectPkg, ".DeepEqual(query.KeySet, ", spannerPkg, ".KeySets()) {")
```

Comment:
Let's add a comment here explaining the pitfall, in case we refactor to ensure this doesn't get lost :sweat_smile: